pulpcore/pulpcore/app/models/fields.py

```diff
@@ -23,6 +27,11 @@ def pre_save(self, model_instance, add):
             Field's value just before saving.
 
         """
+        file_name = str(model_instance.file)
+        if file_name.startswith(os.path.join(settings.MEDIA_ROOT, 'artifact')):
+            raise Exception(_('The file referenced by the Artifact is already present in '
```

Comment:
I think raising base `Exception` should be avoided.  Perhaps `ValueError` would be appropriate.

https://docs.python.org/3/library/exceptions.html#ValueError